### PR TITLE
Move all csproj to .NET 5.0

### PR DESCRIPTION
**Pull request by premun**

### CODE REVIEW

**Comment by premun** [comment]
cc @MaximLipnin

File: src/Microsoft.DotNet.XHarness.iOS.Shared/Microsoft.DotNet.XHarness.iOS.Shared.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <IsPackable>true</IsPackable>
-    <TargetFrameworks>netstandard2.1;netcoreapp3.1</TargetFrameworks>
```
Comment (MattGal) [inline src/Microsoft.DotNet.XHarness.iOS.Shared/Microsoft.DotNet.XHarness.iOS.Shared.csproj:5]: why leave in 2.1 if everything else is up to 5... is this used by a 2.1 project?

File: src/Microsoft.DotNet.XHarness.Tests.Runners/Microsoft.DotNet.XHarness.Tests.Runners.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <IsPackable>true</IsPackable>
-    <TargetFrameworks>netstandard2.1;netcoreapp3.1</TargetFrameworks>
+    <TargetFrameworks>net5.0</TargetFrameworks>
```
Comment (MaximLipnin) [inline src/Microsoft.DotNet.XHarness.Tests.Runners/Microsoft.DotNet.XHarness.Tests.Runners.csproj:5]: It could be `<TargetFramework>`

**Comment by MattGal** [comment]
@premun There may be some desire to continue to target 3.1 for all these projects.  Let's chat when you're available.

**Comment by tkapin** [comment]
Please wait with merging this until we agree on which runtime version to use to run xharness. Based on the conversation with @MattGal  (and @mmitche) it seems it's ok to run on 3.1.x.

File: src/Microsoft.DotNet.XHarness.iOS.Shared/Microsoft.DotNet.XHarness.iOS.Shared.csproj (hunk 1 above)
Comment (premun) [reply src/Microsoft.DotNet.XHarness.iOS.Shared/Microsoft.DotNet.XHarness.iOS.Shared.csproj:5]: iOS.Shared is consumed in .NET 4.8.2 in `xamarin/xamarin-macios`

**Pull request closed without merging**